src/foam/u2/texteditor/ToolBar.js

```diff
@@ -0,0 +1,169 @@
+      display: inline-flex;
+      margin: 16px;
+      padding: 8px 2px;
+      white-space: nowrap;
+
+      /* Firefox */
+      display: -moz-box;
+      -moz-box-pack: center;
+      -moz-box-align: center;
+
+      /* Safari and Chrome */
+      display: -webkit-box;
+      -webkit-box-pack: center;
+      -webkit-box-align: center;
+
+      /* W3C */
+      display: box;
+      box-pack: center;
+      box-align: center;
+    }
+
+    .richtext-actions button{
+      width: auto;
+      height: 40px;
+      text-align: center;
+      font-size: 14px;
+      background: none !important;
+      color: #373a3c !important;
+      border: none;
+      font-weight: normal;
+      border-radius: 0px;
+  
+      -webkit-box-shadow: inset 0 0 0 0 #ffffff;
+    }
+  
+    .richtext-actions button:focus {
+      outline: none;
+      background: #ddd !important;
+    }
+
+   .toolbar-item {
+     width: auto;
+     height: auto;
+     text-align: center;
+     font-size: 14px;
+     background: none !important;
+     color: #373a3c !important;
+     padding: 4px 12px;
+   }  
+
+   .toolbar-menu {
+    position: absolute;
+    background-color: #f0f0f0;
+    min-width: 50px;
+    min-height: 50px;
+    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
+    z-index: 1;
+  }
+
+  .toolbar-menu  button{
+    display: block !important;
+    width: 100px !important;
+    height: 40px;
+    text-align: left !important;
+    padding: 16px;
+    border-radius: 0px;
+  }
+
+ `,
+
+  properties: [
+    {
+      name: 'doc_'
+    }
+  ],
+
+  listeners: [
+    {
+      name: 'refresh',
+      mergeDelay: 300,
+      isMerged: true,
+      code: function() {
+        this.doc_ = null;
+        this.doc_ = this.document;
+        this.refresh();
+      }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this.refresh();
+
+      this.SUPER();
+      this.addClass('richtext-actions');
+      this
+        .start(this.DetailView, { data: this.DefaultTextFormats.create(), showActions: true, title: ''})
+          .addClass('toolbar-item')
+        .end()
+        .start(this.Popup, {
+          button: this.Image.create({ data: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABUAAAAVCAYAAACpF6WWAAAABGdBTUEAALGPC/xhBQAAAAFzUkdCAK7OHOkAAAAgY0hSTQAAeiYAAICEAAD6AAAAgOgAAHUwAADqYAAAOpgAABdwnLpRPAAAAAZiS0dEAAAAAAAA+UO7fwAAAAlwSFlzAAAASAAAAEgARslrPgAAANpJREFUOMvtkyFuw0AQRd+sCiwrICC8zilKAnqIHKAjXyA3aHKDEtN1YY5Q6CP4CAtSKSDAYQXRTohBW7WODYycz3b/7pv/V1qYtKTLLMtyHmPc9IU5595VNTzcODcXkdcBISsguDHqdyYNIRyyLFt+3zOzPfAEfIrI6teV4w9oURSzNE0XHTMaVW2891/t+qKqoTNpkiRrMyv/I5rZDtj2qT/umzrn6hjjDkBEXoBH4Gxmb61fDYaqag3UAN775xba5Hneq/Lo9ScO/fNHmdmHiATgNMS7azxdAX9WQ9Rz12fXAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE5LTAzLTEzVDAzOjE0OjAzKzAwOjAwnW3t/gAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOS0wMy0xM1QwMzoxNDowMyswMDowMOwwVUIAAAAodEVYdHN2ZzpiYXNlLXVyaQBmaWxlOi8vL3RtcC9tYWdpY2stN2xlZW1xSTPhtKVNAAAAAElFTkSuQmCC'}),
+          view: this.DetailView.create({ data: this.FontSize.create(), showActions: true, title: ''})
```

Comment:
Since we'll be using ViewSpecs for the `view` property, you'll need to change this to something like:
```
view: { class: 'foam.u2.DetailView', data: this.FontSize.create() .....
```